### ISSUE TITLE
No transform from [...] to [map]

### ISSUE THREAD

**Issue by EnderDragonEP** [linorobot/linorobot2#48]
I run:
`ros2 launch linorobot2_bringup bringup.launch.py`
`ros2 run teleop_twist_keyboard teleop_twist_keyboard`
`ros2 launch linorobot2_navigation slam.launch.py rviz:=true`
to slam, but I the transform wasn't working. And rviz has a lot of errors displayed all related to transform.
![image](https://user-images.githubusercontent.com/30905525/191463228-9f916e63-73bb-4ec4-8002-93937d7cebad.png)
![image](https://user-images.githubusercontent.com/30905525/191463265-35e170d8-f8d5-4468-95f0-3a20e65851f5.png)
![image](https://user-images.githubusercontent.com/30905525/191463322-aa0578ae-d48e-492b-8ee1-d151b374e7c3.png)

I'm new to ROS2 and wasn't able to Identify and fix the problems... 

URDF looks ok and working:
![image](https://user-images.githubusercontent.com/30905525/191464969-785ee445-07b1-4594-a219-7e3c9a979625.png)

Topic list:
```
/clicked_point
/cmd_vel
/diagnostics
/goal_pose
/imu/data
/initialpose
/joint_states
/map
/map_metadata
/map_updates
/odom
/odom/unfiltered
/parameter_events
/robot_description
/rosout
/scan
/set_pose
/slam_toolbox/feedback
/slam_toolbox/graph_visualization
/slam_toolbox/scan_visualization
/slam_toolbox/update
/tf
/tf_static
```

* System info
  System config: 2wd
  System: Jetson nano 
  OS: Ubuntu 20.04 from [Qengineering](https://github.com/Qengineering/Jetson-Nano-Ubuntu-20-image)
  Base board: Teency 4.0
  Lidar: Rplidar
  IMU:  MPU-9250
  Motor driver: L298N

**Comment by grassjelly** [comment]
Have you initialized the robot’s pose in rviz?

**Comment by EnderDragonEP** [comment]
> Have you initialized the robot’s pose in rviz?

Thanks for the advice! I'll try.
But  SLAM need the position of the robot? That's kinda different from Linorobot revision one isn't it?

**Comment by grassjelly** [comment]
Sorry, I missed the slam part. Are you able to confirm that these topics https://github.com/linorobot/linorobot2_hardware#3-check-the-topics are available and publishing topics?

**Comment by EnderDragonEP** [comment]
> Sorry, I missed the slam part. Are you able to confirm that these topics https://github.com/linorobot/linorobot2_hardware#3-check-the-topics are available and publishing topics?

Yes they are available odometry and IMU data are present.
![image](https://user-images.githubusercontent.com/30905525/191699928-a387a5bf-2cf1-469c-8149-bf00ea3ffeae.png)

/odom/unfiltered
![image](https://user-images.githubusercontent.com/30905525/191700510-664dfb86-e461-472d-8de5-472f1af66946.png)

/imu/data
![image](https://user-images.githubusercontent.com/30905525/191700668-7ed7056f-34e2-4060-9a74-2d328f674993.png)

**Comment by grassjelly** [comment]
Silly question, how about the `/scan` topic?

**Comment by EnderDragonEP** [comment]
> Silly question, how about the `/scan` topic?

Thank you for the head-up! I have found the cause of the problem. 
I was thinking about where is the laser scan lines goes and I was checking the [manual installation](https://github.com/linorobot/linorobot2/blob/galactic/ROBOT_INSTALLATION.md).
I installed the **Rplidar** and **Realsense** drivers and add the environment variables to the `.bashrc`.
Out of curiosity, I opened the `.bashrc` and I find that the  `LINOROBOT2_DEPTH_SENSOR` and `LINOROBOT2_LASER_SENSOR` variables are not set.
![image](https://user-images.githubusercontent.com/30905525/191715681-20977f41-f10c-4a50-8cea-c36c6149e86d.png)

After I eddit the `.bashrc` and run the command `source ~/.bashrc`. The issue was fixed! No more errors.
![image](https://user-images.githubusercontent.com/30905525/191718035-c11f59d9-9893-4d37-9e5d-64ce7c7ef818.png)

But seems like the Intel Realsense camera still have some issue. But that's the problem for another day. 😭
![image](https://user-images.githubusercontent.com/30905525/191718494-e5ab81af-1299-4d39-ba86-f073a6890fc1.png)

Thank for you help and guidance!
